### PR TITLE
Redirect to email confirmation screen when email has not been confirmed yet

### PR DESCRIPTION
**Pull request by Devessier**
This PR adds the redirection to email confirmation screen after application boot, signing in and signing up, if the user has not yet confirmed her email.
The next step is to add a form to actually confirm email.

### CODE REVIEW

File: packages/client/machines/appMachine.ts [hunk 1]
```diff
@@ -271,6 +270,25 @@ export function createAppMachine({
                     },
                 },
 
+                checkingUserAccountValidity: {
```
Comment (prastoin) [inline packages/client/machines/appMachine.ts:273]: praise: great !

**Review by prastoin — APPROVED** [review]
Really nice work !
So smooth ready to be merged

File: packages/client/navigation/index.tsx [hunk 2]
```diff
@@ -164,6 +169,12 @@ export const RootNavigator: React.FC<ColorModeProps> = ({ colorScheme }) => {
                         options={{ headerShown: false }}
                     />
                 </>
+            ) : applicationState === 'EMAIL_NOT_CONFIRMED' ? (
```
[the remark this answers is not in the document]
Comment (Devessier) [reply packages/client/navigation/index.tsx:172]: Indeed, I forgot about that!
We thought about letting the user going back to signing in and signing out screens on Basecamp:
e6b29e2e-a9fc-11ec-8c36-4e8159c9eff9

We can decide what we really want to do later.

[the remark this answers is not in the document]
Comment (prastoin) [reply packages/client/navigation/index.tsx:172]: Yes globally I think we will have to display the sign IN/UP and email confirmation screen inside the same instance ??
I don't think that a goback button makes sense haha, or maybe the go back button would perform a signOut ?

**Pull request merged**